lib/phoenix/test/conn_test.ex

```diff
@@ -410,7 +410,7 @@ defmodule Phoenix.ConnTest do
       assert "can't be blank" in body["errors"]
 
   """
-  @spec json_response(Conn.t, status :: integer | atom) :: map | no_return
+  @spec json_response(Conn.t, status :: integer | atom) :: map | list | no_return
```

Comment:
to be fair, other types can be there as well, like string, nil, integer, float, boolean

Reply:
You are totally right. So I went through JSON spec https://www.json.org/json-en.html and thankfully apart from arrays (lists) and objects (maps) there are only there are only a couple of primitives:
```
string - String.t
number - integer | float
true | false - boolean
null - nil
```
That should be exhaustive.

I was wandering about stealing it from Jason, but actually, Jason uses `term`...
Maybe we shouldn't be so granular and also return `term`?

Reply:
`Poison` uses more precise definition:

https://github.com/devinus/poison/blob/de07407fd35293d136ef3c7b2a140eaa29ff5617/lib/poison/parser.ex#L69-L75

Let's wait what maintainers say.
But I'd use this definition:
```
@type json :: boolean() | nil | float() | integer() | String.t() | [json()] | %{String.t() => json()}
```

Reply:
I've thought about it some more and I've changed it to `term`.
For e.g. `Jason` allows decoding floats either to `native` or to `Decimal` so that type would not be covered by the above definition.
We would need to add every decoding possibility from every library to cover all edge cases and make dialyzer happy.
I think it is better to go back to a more generic `term`.

Thanks, @fuelen for the fruitful discussion!

Reply:
Converting to `Decimal` is possible only if `decode/2` is used, which is not the case